docker/utils/utils.py

```diff
@@ -445,26 +445,45 @@ def parse_devices(devices):
 
 def kwargs_from_env(ssl_version=None, assert_hostname=None):
     host = os.environ.get('DOCKER_HOST')
-    cert_path = os.environ.get('DOCKER_CERT_PATH')
+
+    # empty string for cert path is the same as unset.
+    cert_path = os.environ.get('DOCKER_CERT_PATH') or None
+
+    # empty string for tls verify counts as "false".
+    # Any value or 'unset' counts as true.
     tls_verify = os.environ.get('DOCKER_TLS_VERIFY')
+    if tls_verify == '':
+        tls_verify = False
+        enable_tls = True
+    else:
+        tls_verify = tls_verify is not None
+        enable_tls = cert_path or tls_verify
```

Comment:
If I understand correctly that would enable my use case, so I'd be fine with that. However it seems like it would be better to enable verification by default and require an explicit setting to turn it off (specifically, setting DOCKER_TLS_VERIFY to a falsy value).

Also you don't need the trailing `or False`, it would be more pythonic to write your snippet as:

```python
tls_verify = os.environ.get('DOCKER_TLS_VERIFY')
enable_tls = tls_verify or cert_path
```

or perhaps wrap the values in `bool` such as `bool(os.environ.get('DOCKER_TLS_VERIFY'))`